ast/compile_test.go

```diff
@@ -5053,6 +5053,82 @@ func TestRewriteDeclaredVars(t *testing.T) {
 	}
 }
 
+func TestCompileUnusedAssignedVarsErrorLocations(t *testing.T) {
+	tests := []strictnessTestCase{
+		{
+			note: "one of the two function args is not used - issue 5602 regression test",
+			module: `package test
+			func(x, y) {
+				x = 1
+			}`,
+			expectedErrors: Errors{
+				&Error{
+					Code:     CompileErr,
+					Location: NewLocation([]byte("x = 1"), "", 3, 5),
```

Comment:
I'd expected the location of the error to be the head of the function, and not the first line of its body.
This seems to be a general issue for this error type, though, where we always use the first line of the body.

https://github.com/open-policy-agent/opa/blob/ca72072a1f7466afde41b9f82a1067861f3b4fbf/ast/compile.go#L4731

I think it's ok to treat this as a separate issue that doesn't need to be solved in this PR.

Reply:
https://github.com/open-policy-agent/opa/issues/5662